gateway-server/src/main/java/org/apache/knox/gateway/services/token/impl/AliasBasedTokenStateService.java

```diff
@@ -166,24 +166,21 @@ protected long getMaxLifetime(final String tokenId) {
 
   @Override
   public long getTokenExpiration(String tokenId, boolean validate) throws UnknownTokenException {
-    long expiration = 0;
-
-    if (!validate) {
-      // If validation is not required, then check the in-memory collection first
-      try {
-        expiration = super.getTokenExpiration(tokenId, validate);
-        return expiration;
-      } catch (UnknownTokenException e) {
-        // It's not in memory
-      }
+    if (validate) {
```

Comment:
I would keep this where it was previously, since the super will validate based on the in-memory store, and if it's not in-memory, only then do we want to validate the alias.

Reply:
Fixed.